csrc/quantization/compressed_tensors/int8_quant_kernels.cu

```diff
@@ -0,0 +1,51 @@
+#include <ATen/cuda/CUDAContext.h>
+#include <torch/extension.h>
+
+#include "../../dispatch_utils.h"
+
+static inline __device__ int8_t float_to_int8_rn(float x) {
+  static constexpr float dt_min =
+      static_cast<float>(std::numeric_limits<int8_t>::min());
+  static constexpr float dt_max =
+      static_cast<float>(std::numeric_limits<int8_t>::max());
+  // round
+  float dst = round(x);
```

Comment:
Note - This rounding doesn't match Compressed-tensors's/Torch's/Numpy's rounding method. To fix this I have a patch at https://github.com/neuralmagic/nm-vllm/pull/263 - this will be merged before this lands.

Reply:
Fixed!